addons/web/static/src/model/relational_model/relational_model.js

```diff
@@ -223,6 +236,30 @@ export class RelationalModel extends Model {
         }
     }
 
+    /**
+     * If we group by default based on a property, the property might not be loaded in `fields`.
+     */
+    async _getPropertyDefinition(config, propertyFullName) {
+        // dynamically load the property and add the definition in the fields attribute
+        const result = await this.orm.call(
+            config.resModel,
+            "get_property_definition",
+            [propertyFullName],
+            { context: config.context }
+        );
+        if (!result) {
+            // the property might have been removed
+            config.groupBy = null;
```

Comment:
Note that this code also applies in list view, where you can have several level on groupbys, in this case, you remove all the groupbys.

Also, you can groupby several fields, and if there's a property among them, but not as first groupby, your code won't be executed.

Reply:
> Note that this code also applies in list view, where you can have several level on groupbys, in this case, you remove all the groupbys.

This is ok I think, it's already a edge case where you have a default group by, and you remove the property (the user will have to re-do the group by anyway)

Reply:
What about
 
> Also, you can groupby several fields, and if there's a property among them, but not as first groupby, your code won't be executed.

?

Reply:
it work, if you save a group by with a property as second, the property definition will be fetched when you unfold the list

![image](https://github.com/odoo/odoo/assets/54316226/170ea98e-7fcc-4c71-bf8d-6fadc953f4be)